materialhelptutorial/src/main/java/za/co/riggaroo/materialhelptutorial/MaterialTutorialFragment.java

```diff
@@ -65,9 +66,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         if (tutorialItem.getBackgroundImageRes() != -1) {
             Glide.with(this).load(tutorialItem.getBackgroundImageRes()).into(imageViewBack);
         }
-        if (tutorialItem.getForegroundImageRes() != -1) {
+        if (tutorialItem.getForegroundImageRes() != -1 && !tutorialItem.isGif()) {
             Glide.with(this).load(tutorialItem.getForegroundImageRes()).into(imageViewFront);
         }
+        if (tutorialItem.getForegroundImageRes() != -1 && tutorialItem.isGif()){
+            Glide.with(this).load(tutorialItem.getForegroundImageRes()).asGif().diskCacheStrategy(DiskCacheStrategy.SOURCE).into(imageViewFront);
```

Comment:
I think the best would be to use `NONE` for **all** loads, after all this should be only seen once by each user, and if the user goes back to a previous page of the tutorial it should still be in memory cache. After it is dismissed the files will just stay in cache otherwise.